### DISCUSSION TITLE
GUIs

### DISCUSSION THREAD

**Discussion by spikefishjohn** [GNS3/gns3-gui#3342] [General]
Are there any plans to sunset the fat client? I'm just wondering where things are going. I would think at some point the fat client would get depreciated so that a given feature isn't developed twice. 

For example, the fat client can set a MAC of a given NIC but the webui seems like it can't. Just seems like there aren't many development resources behind GNS3 so I'm wondering if it makes sense to continue supporting two GUIs.

**Comment by grossmj** [comment]
> Are there any plans to sunset the fat client? I'm just wondering where things are going. I would think at some point the fat client would get depreciated so that a given feature isn't developed twice.

We don't have any immediate plans to sunset the fat client but we will consider it once we are fully satisfied with the web interface and have (almost) all the features that the fat client has. I am also concerned that some users may prefer a fat client over a web based one, so we may run some polls once we have that rock solid web-ui.

> For example, the fat client can set a MAC of a given NIC but the webui seems like it can't. Just seems like there aren't many development resources behind GNS3 so I'm wondering if it makes sense to continue supporting two GUIs.

This example is one of the many small things that are yet to make it to the web-ui. We recently got a dedicated dev for the web-ui (@rajnikantsolarwinds) so hopefully things will move faster on this side. One thing for sure is we won't put too much work in the fat client anymore and focus most of our time on the back-end and the web-ui.

**Comment by spikefishjohn** [comment]
I run GNS3 on a Dell R720. The iDac on the R720 has a built in html5 VNC client. The R720 came out 10 years ago.

Granted.. dell is most likely funded better then GNS3.

Is the webui's main code repo somewhere else? It seems like no major work has been done since Feb on the webui.

**Reply by grossmj** [reply]
The web-ui registry is on https://github.com/gns3/gns3-web-ui and most of the work is on the master-3.0 branch

**Comment by spikefishjohn** [comment]
https://github.com/GNS3/gns3-server/pull/2077

This seems to make it so I can deploy a node on to a remote server using a template stored on the main server. This doesn't address that the UI won't let you set the compute_id field to null. I used curl .. or maybe anible to do that. Honestly can't remember. 

```</ComicBookGuy> Worst place for a bug report ever.```

**Reply by grossmj** [reply]
The UI v3 will support a null compute_id: https://github.com/GNS3/gns3-gui/issues/3265

**Comment by kefins** [comment]
Hi, when will 3.0 be released, was there a plan or a roadmap?
